projects/observability/src/shared/components/cartesian/d3/legend/cartesian-legend.ts

```diff
@@ -133,4 +193,33 @@ export class CartesianLegend {
       })
     );
   }
+
+  private makeSelectionModeOff(): void {
+    this.activeSeries = [...this.initialSeries];
+    this.isSelectionModeOn = false;
+    this.updateLegendClassesAndStyle();
+    this.setResetVisibility(!this.isSelectionModeOn);
+    this.activeSeriesSubject.next();
+  }
+
+  private updateActiveSeries(seriesEntry: Series<{}>): void {
+    if (!this.isSelectionModeOn) {
+      this.activeSeries = [];
+      this.activeSeries.push(seriesEntry);
```

Comment:
nit - why reset then mutate?
```suggestion
      this.activeSeries = [seriesEntry];
```